app/src/main/java/org/xwiki/android/sync/ViewModel/BaseViewModelFactory.kt

```diff
@@ -0,0 +1,10 @@
+package org.xwiki.android.sync.ViewModel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+class BaseViewModelFactory<T>(val creator: () -> T) : ViewModelProvider.Factory {
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
```

Comment:
I am not sure that it is good idea to use `modelClass` here? I think, it is better to provide special objects for any required class